### PR TITLE
fix:html to node converter fixes

### LINKED ISSUE #139: [Migration]  Migrate Html To Node Converter to latest changes
### Bug Description

converting html to node the documents are not rendered by the components

### How to Reproduce

Use Test Raw html
 ```
 final editorState =
              EditorState(document: HTMLToNodesConverter(rawHTML).toDocument());
```
               
Screenshot 2023-05-29 at 11 13 11 AM

### Expected Behavior

Document Should be rendered

### Operating System

macos

### AppFlowy Editor Version(s)

latest main

### Screenshots

_No response_

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by alihassan143**
#139 
need help in finding that repeating text is already added or not that add attribute on that text
Screenshot 2023-05-31 at 6 00 15 PM
current progress

```
<p>&nbsp;</p>
      <h1><span style="text-decoration: underline;"><em>AppFlowy</em></span> <em><strong>Editor</strong></em> is a <strong><em>highly</em> customizable</strong> <em>rich-text editor</em> for&nbsp; <em><strong>Flutter</strong></em></h1>
      <ol>
      <li>You <em>can</em> <strong>also</strong> use <span style="background-color: rgba(0, 188, 240, 96); font-weight: bold; font-style: italic;">AppFlowy Editor</span> as a component to build your own app.</li>
      <li>ertteterttet</li>
      <li>If you have questions or <strong>feedback</strong>, please submit an issue on Github or join the community along with 1000+ builders!</li>
      </ol>
      <ul>
      <li>You&nbsp;<em>can</em>&nbsp;<strong>also</strong>&nbsp;use&nbsp;<span style="background-color: rgba(0, 188, 240, 96); font-weight: bold; font-style: italic;">AppFlowy Editor</span>&nbsp;as a component to build your own app.</li>
      <li>ertteterttet</li>
      <li>&nbsp;</li>
      <li><strong>abdibabdjnaodn</strong></li>
      </ul>
      <p><em>andkajskndljsadl</em></p>
      <p>&nbsp;</p>
      <p><em><strong><span style="text-decoration: underline;">bnaksdnlasdnlasnldnl.&nbsp; &nbsp;</span></strong></em></p>
      <p><em><strong><span style="text-decoration: underline;">asoidasjda</span></strong></em></p>
      <p><em><strong><span style="text-decoration: underline;">ashjdjasdnkasdad</span></strong></em></p>
      <p><em><strong><span style="text-decoration: underline;">knajsdlasdlad</span></strong></em></p>
      <h1><span style="text-decoration: underline;"><span style="text-decoration: underline;"><em>AppFlowy</em></span>&nbsp;<em>Editor</em>&nbsp;is a&nbsp;<strong><em>highly</em>&nbsp;customizable</strong>&nbsp;<em>rich-text editor</em>&nbsp;for&nbsp;&nbsp;<em>Flutter</em></span></h1>
      <ol>
      <li><span style="text-decoration: underline;">You&nbsp;<em>can</em>&nbsp;<strong>also</strong>&nbsp;use&nbsp;<span style="background-color: rgba(0, 188, 240, 96); font-weight: bold; font-style: italic;">AppFlowy Editor</span>&nbsp;as a component to build your own app.</span></li>
      <li><span style="text-decoration: underline;">ertteterttet</span></li>
      <li><span style="text-decoration: underline;">If you have questions or&nbsp;<strong>feedback</strong>, please submit an issue on Github or join the community along with 1000+ builders!</span></li>
      </ol>
```
Html Code

### CODE REVIEW

**Comment by alihassan143** [comment]
Screenshot 2023-05-31 at 6 04 19 PM
Html Editor View

**Comment by LucasXu0** [comment]
Hey, @alihassan143. Please add tests for it.

**Comment by alihassan143** [comment]
@LucasXu0  tests added

**Comment by alihassan143** [comment]
@LucasXu0  now pr ready for review test added editor rendering as expected

**Comment by LucasXu0** [comment]
Awesome. Let me check it now.

File: lib/src/plugins/html/html_document_decoder.dart [hunk 1]
```diff
@@ -46,36 +52,48 @@ class DocumentHTMLDecoder extends Converter<String, Document> {
...
+    } else if (element.localName == HTMLTag.image) {
+      return [_handleImage(element)];
+    } else if (element.localName == HTMLTag.blockQuote) {
+      return [_parseBlockQuoteElement(element)];
+    } else if (delta != null) {
+      Attributes attributes = _getParserFormattingElement(element);
+      delta.insert(element.text, attributes: attributes);
+      return [];
+    } else {
+      return [paragraphNode(delta: Delta()..insert(element.text))];
```
Comment (LucasXu0) [inline lib/src/plugins/html/html_document_decoder.dart]: I have no idea why you changed the switch case to if else statement。

Comment (alihassan143) [reply lib/src/plugins/html/html_document_decoder.dart]: if we add switch case it starts missing variables and we cannot add delta conditions in it

**Comment by LucasXu0** [comment]
I refactored some part of them. will merge it once the CI pass.

**Comment by alihassan143** [comment]
> I refactored some part of them. will merge it once the CI pass.

ok

File: lib/src/plugins/html/html_document_decoder.dart (hunk 1 above)
Comment (LucasXu0) [reply lib/src/plugins/html/html_document_decoder.dart]: Actually, we don't need to handle delta here. I moved this logic to another function.

Comment (alihassan143) [reply lib/src/plugins/html/html_document_decoder.dart]: ok

**Comment by alihassan143** [comment]
@LucasXu0  test passed but their are conflicts issue

**Comment by LucasXu0** [comment]
No worries. I have solved them.

**Comment by alihassan143** [comment]
@LucasXu0  we need to start delete unused files in the repo after changes

**Comment by alihassan143** [comment]
@LucasXu0  code coverage issue

**Comment by LucasXu0** [comment]
> @LucasXu0 we need to start delete unused files in the repo after changes

I know. I plan to remove all the unused files before releasing the stable version.

**Comment by alihassan143** [comment]
@LucasXu0  i added android and ios in yaml file inside platforms

**Comment by alihassan143** [comment]
#148

**Comment by LucasXu0** [comment]
@alihassan143 then please submit a new PR.

**Comment by alihassan143** [comment]
@LucasXu0  i added in this pr if you want me to create new pr than i will do that

**Comment by LucasXu0** [comment]
LGTM.

**Pull request merged**